src/main/java/org/checkerframework/specimin/UnsolvedClass.java

```diff
@@ -108,6 +108,26 @@ public void updateMethodByReturnType(
     }
   }
 
+  /**
+   * This method updates the types of fields in this class
+   *
+   * @param currentType the current type
+   * @param correctType the desired type
+   */
+  public void updateFieldByType(String currentType, String correctType) {
+    for (String fieldExpression : classFields) {
+      String[] elements = fieldExpression.split(" ");
```

Comment:
What if the field has a type annotation on the type? Specimin needs to preserve type annotations (do we have a test for that?); if there is one, I'd expect more than one space.

For example, what if the original field declaration is something like `@Nullable Foo foo;`?

Reply:
I don't expect to see any annotation here, since these fields are inside synthetic classes. Synthetic classes are created by Specimin for unsolved symbols. These symbols are unsolved because the corresponding class files do not exist in the root directory physically. Specimin will create synthetic versions for these classes so that we will end up with a minimized version of the input codes that can be compiled.

Reply:
So if this is the input codes:

```
package org.example;

class Child extends Parent {
  public @NonNull String getLastName() {
     return lastName;
   }
}
```
Then the synthetic class for the Parent class will be: 
```
package org.example;

class Parent{
  public String lastName = null;
}
```

As I understand, it's the job of the user to look that this result and realize that the field `lastName` in the real class file (not the synthetic file) of the Parent class must not be null. So he must look up the codes of the Parent class to make sure that the field `lastName` is not null so that his type reasoning can be valid.

Reply:
Thanks, this makes sense. Can you add a comment at this point in the code that reminds the reader that `fieldExpression` is guaranteed not to have an annotation, because it's part of a synthetic class? That's why it is safe to assume that the result of `split` has exactly two elements.